### PR TITLE
Feature/develop better host name

### PR DESCRIPTION
**Pull request by magicprinc**
Same https://github.com/osiegmar/logback-gelf/issues/90
but for develop branch

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties
Comment (osiegmar) [inline gradle/wrapper/gradle-wrapper.properties:1]: docs and sources are unnecessary 99% of the time and cause unnecessary traffic

File: src/main/java/de/siegmar/logbackgelf/GelfEncoder.java [hunk 1]
```diff
@@ -410,13 +413,13 @@ public byte[] encode(final ILoggingEvent event) {
 
         final byte[] json = gelfMessageToJson(gelfMessage);
 
-        if (appendNewline) {
-            final byte[] sep = System.lineSeparator().getBytes(StandardCharsets.UTF_8);
-            final ByteBuffer bb = ByteBuffer.allocate(json.length + sep.length);
+        if (appendNewline){
+            final byte sep = '\n';//System.lineSeparator().getBytes(StandardCharsets.UTF_8);
+            final ByteBuffer bb = ByteBuffer.allocate(json.length + 1);// sep.length
```
Comment (osiegmar) [inline src/main/java/de/siegmar/logbackgelf/GelfEncoder.java:418]: I assume this breaks windows console logging – the feature this was built for...

File: src/main/java/de/siegmar/logbackgelf/GelfEncoder.java [hunk 2]
```diff
@@ -398,6 +398,9 @@ public byte[] encode(final ILoggingEvent event) {
             shortMessage = "Empty message replaced by logback-gelf";
             addWarn("Log message was empty - replaced to prevent Graylog error");
         }
+        if (shortMessage.length() > MAX_SHORT_MESSAGE_LENGTH) {
+            shortMessage = shortMessage.substring(0, MAX_SHORT_MESSAGE_LENGTH - 1);
+        }
```
Comment (osiegmar) [inline src/main/java/de/siegmar/logbackgelf/GelfEncoder.java:403]: Why?

File: src/main/java/de/siegmar/logbackgelf/GelfMessage.java [hunk 3]
```diff
@@ -29,13 +27,15 @@
...
+import static java.nio.charset.StandardCharsets.*;
+
 /**
  * Class for GELF 1.1 format representation.
  */
 public class GelfMessage {
 
     private static final String VERSION = "1.1";
-    private static final int INITIAL_SIZE = 256;
+    private static final int INITIAL_SIZE = 502;
```
Comment (osiegmar) [inline src/main/java/de/siegmar/logbackgelf/GelfMessage.java:38]: Why? Is 256 to low for average use? Did you mean 512 instead?

File: src/main/java/de/siegmar/logbackgelf/InetUtil.java [hunk 4]
```diff
@@ -20,13 +20,41 @@
...
 
 final class InetUtil {
 
     private InetUtil() {
     }
 
+    /**
+     * Set this system property to a simple hostname to use a fixed host name.
+     */
+    public static final String PROPERTY_LOGSTASH_GELF_HOSTNAME = "logstash-gelf.hostname";
```
Comment (osiegmar) [inline src/main/java/de/siegmar/logbackgelf/InetUtil.java:36]: This library is named `logback-gelf` 😉

File: src/test/java/de/siegmar/logbackgelf/InetUtilTest.java
Comment (osiegmar) [inline src/test/java/de/siegmar/logbackgelf/InetUtilTest.java:1]: unrelated

File: src/main/java/de/siegmar/logbackgelf/InetUtil.java [hunk 5]
```diff
@@ -20,13 +20,41 @@
...
+        return null;
+    }
+
+    static String trim (String str){
+        return str == null ? ""
+            : str.trim().strip();
+    }
+
     /**
      * Retrieves the local host's name. Preferably the fully qualified one.
```
Comment (osiegmar) [inline src/main/java/de/siegmar/logbackgelf/InetUtil.java:59]: Javadoc doens't seem to match the implementation now. Tests are missing.

**Review by osiegmar — CHANGES_REQUESTED** [review]
Thanks for your contribution! Mixing unrelated things in one PR makes the review unnecessary cumbersome. Please split in purpose-oriented PRs, add missing documentation / tests and fix the static code analysis errors.

File: src/main/java/de/siegmar/logbackgelf/GelfEncoder.java (hunk 1 above)
Comment (magicprinc) [reply src/main/java/de/siegmar/logbackgelf/GelfEncoder.java:418]: Windows Console supports both \r\n and \n
Please try yourself.
Not only System.out.print("line1\nline2\r\nline3") works, but also $ type FileWithLfEols.java

File: src/main/java/de/siegmar/logbackgelf/GelfEncoder.java (hunk 2 above)
Comment (magicprinc) [reply src/main/java/de/siegmar/logbackgelf/GelfEncoder.java:403]: Full compatibility with your closest competitor :-)

File: src/main/java/de/siegmar/logbackgelf/GelfMessage.java (hunk 3 above)
Comment (magicprinc) [reply src/main/java/de/siegmar/logbackgelf/GelfMessage.java:38]: No. Default packet size is 502 (see your UDP sources).
We have long logs, so usually we fill the whole packet, so it makes sense to make less byte[] reallocations. Not critical, but also no harm. 256 looks like completely abstract number (2⁸) too.

File: src/main/java/de/siegmar/logbackgelf/InetUtil.java (hunk 5 above)
Comment (magicprinc) [reply src/main/java/de/siegmar/logbackgelf/InetUtil.java:59]: This is copt paste from my working tested code. There is additional test, you named it "unrelated" :-)

Your original compact version is broken in case of Docker installed: it shows not a real hostname, but something like internal.docker.host

The new version also supports explicit hostname specification (through system property)

File: src/main/java/de/siegmar/logbackgelf/GelfEncoder.java (hunk 1 above)
Comment (osiegmar) [reply src/main/java/de/siegmar/logbackgelf/GelfEncoder.java:418]: Still don't see a reason to change this. Windows line ending is \r\n even if the terminal now supports unix line endings.

**Review by osiegmar — COMMENTED** [review]
Thanks for pointing out the windows build problem. I decided to fix it by 819ee77499433d030cc22b2216668009574b7de9 and added a Windows build runner configuration to this project.

**Comment by osiegmar** [comment]
Thanks for your contribution!
